### PR TITLE
Dark theme support

### LINKED ISSUE #210: Dark theme
It would be cool if we had a dark theme toggle button, and even better if dark/bright color scheme preference was detected automatically. I might work on this if I have some time.

Congrats for getting the project back on track!

### PR DESCRIPTION
**Pull request by Arkelis**
Resolve #210

Some other things to do before merging:
- [x] Dark colors for `index.html` 
- [x] Dark colors for sidebar
- [x] Update doc
- [x] Update `customtemplate/module.html.jinja2` template for dark theme only example
- [x] Remove {{ pygment_css }} when not needed

<details><summary>Overview</summary>
<p>

</p>
</details>

### CODE REVIEW

File: pdoc/templates/default/module.html.jinja2 [hunk 1]
```diff
@@ -31,6 +70,21 @@
...
+                      color: var(--main-text-color);
+                      background-color: var(--main-bg-color);
+                }
+
+                 @media (prefers-color-scheme: dark) {
+                      body {
+                          --main-bg-color: #212529;
+                          --main-text-color: #fff;
+                      }
+                 }
```
Comment (mhils) [inline pdoc/templates/default/module.html.jinja2:86]: Let's move the body colors specifications to the rest of the colors specs and can combine the two `@media (prefers-color-scheme: dark)` queries.

**Review by mhils — COMMENTED** [review]
This looks pretty excellent, thanks!

I've been toying around with your implementation for a bit, looks great. I think I also found a way to make it easily possible for users to enable dark mode by default:

`default/module.html.jinja2`
```html+jinja

  {% block default_theme %}
    .pdoc { ... }
    body { ... }
  {% endblock %}
  @media (prefers-color-scheme: dark) {
    {% block dark_theme %}
      .pdoc { ... }
      body { ... }
    {% endblock %}
  }
```
custom `module.html.jinja2`
```html+jinja
{% extends "default/module.html.jinja2" %}
{% block default_theme %}
    {{ self.dark_theme.super() }}
{% endblock %}
{% block dark_theme %}{% endblock %}
```

That still needs a custom code-highlighting.css, but I think that's a fair tradeoff 😃 Do you want to add this to the https://github.com/mitmproxy/pdoc/blob/main/test/customtemplate/module.html.jinja2 example as well?

File: pdoc/templates/code-highlighting.css [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+ *
+ *     pygmentize -S <theme> -f html > code-highlighting.css
+ *
+ * You can then append to this file an alternate dark theme with the following:
+ *
+ *     echo "\n@media(prefers-color-scheme: dark){ \
+             \n$(pygmentize -S <theme> -f html)\n}" >> code-highlighting.css
+ *
+ * You can test different builtin themes at https://pygments.org/demo/
+ */
```
[the remark this answers is not in the document]
Comment (Arkelis) [reply pdoc/templates/code-highlighting.css:4]: Done in b444290

File: pdoc/templates/default/module.html.jinja2 (hunk 1 above)
Comment (Arkelis) [reply pdoc/templates/default/module.html.jinja2:94]: Done in ceb59b0

File: pdoc/templates/README.md [hunk 3]
```diff
@@ -9,3 +9,22 @@ a `module.html.jinja2` file in your custom template directory that extends the d
...
+```
+
+Then, you can add a dark color scheme:
+
+```
+echo "\n@media(prefers-color-scheme: dark){ \
+      \n$(pygmentize -S <theme> -f html)\n}" >> code-highlighting.css
+```
+
+You can test different builtin themes on [the Pygments website](https://pygments.org/demo/).
```
[the remark this answers is not in the document]
Comment (Arkelis) [reply pdoc/templates/README.md:31]: Done in b444290 :)

**Comment by Arkelis** [comment]
I just added the dark colors for `index.html.jinja2`, and I was wondering if it could not be simpler to specify the color variables (the two blocks that you suggested: `{% block default_theme %}` and `{% block dark_theme %}` in `frame.html.jinja2`), in order to centralize theming tweaking. What do you think?

**Comment by mhils** [comment]
The problem with `frame.html.jinja2` is that it can only be replaced, not extended. We do this "use the user-defined template, fallback to default/ if it doesn't exist" pattern here:
https://github.com/mitmproxy/pdoc/blob/cfbe609b1cd48f568fbd87926d6199e160d73c97/pdoc/render.py#L68-L70

This allows users to `{% extends "default/..." %}` in their override.

Unfortunately, as far as I know there is nothing similar for the `extends` statement:
https://github.com/mitmproxy/pdoc/blob/cfbe609b1cd48f568fbd87926d6199e160d73c97/pdoc/templates/default/module.html.jinja2#L1

Long story short, putting more stuff into `frame.html.jinja2` is bad idea. Let's just keep the styles separated. :)

**Comment by Arkelis** [comment]
There are some merge commits because I forked and commited before you reworked CSS, if you want I can try to do a clean rebase

**Comment by mhils** [comment]
No, that's fine - thanks! 😃 I'll get to this tomorrow probably.

**Comment by mhils** [comment]
Excellent work! I have merged your work in 57b39bbc6dd90b525991ea30b0e58cec349a3e16 and then reduced the number of variables a bit in b384adb05bf3bf5cdbff71cd379a6b207a20276f. I have just pushed 6.0 including all changes. :)

After playing around with this for a bit longer I have decided to take out the `prefers-color-scheme` media query from the default template - it is just too much of a footgun when embedding pdoc into another website, which then doesn't render properly for users with dark mode enabled. Instead, I have moved dark mode to [examples/dark-mode](https://github.com/mitmproxy/pdoc/tree/main/examples/dark-mode), where the emphasis is now on setting a dark mode as the default for all users. 🌙 

Thanks again for the fantastic PR and the quick turnaround time! 🍰

**Comment by Arkelis** [comment]
Great! Thanks for merging :)
